reflector.go

```diff
@@ -114,8 +114,21 @@ func (r *Reflector) Reflect(mappings []Mapping) error {
 				},
 			},
 			Data: k8sSecretData,
+			Type: v1.SecretTypeOpaque,
 		}
 
+		// if the secret has ".dockercfg", use type "kubernetes.io/dockercfg"
+		if k8sSecretData[v1.DockerConfigKey] != nil {
+			newSecret.Type = v1.SecretTypeDockercfg
+		}
+
+		// same with .dockerconfigson
+		if k8sSecretData[v1.DockerConfigJsonKey] != nil {
```

Comment:
Why not `else if` or swich/case?

Reply:
I started with that, and then started thinking about priority ordering, and both ways have opposite concerns. If/Else also doesn't solve the case when you have both fields. One is going to win unless we explicitly looks for that (not worth it).

I took it out of the else mostly so the comments lined up, and I thought it was a little more aesthetically pleasing. If you feel strongly one way or another I can change it.